src/Type/CommonUnionType.php

```diff
@@ -122,4 +123,19 @@ public function changeBaseClass(string $className): StaticResolvableType
 		return new self(UnionTypeHelper::changeBaseClass($className, $this->getTypes()));
 	}
 
+	public function isIterable(): int
+	{
+		return self::RESULT_NO;
+	}
+
+	public function getIterableKeyType(): Type
+	{
+		return new MixedType();
```

Comment:
I'd say ErrorType. In all places where isIterable says NO, the related methods should return ErrorType.